### ISSUE TITLE
pre-generator is not disabling

### ISSUE THREAD

**Issue by ohxorud-dev** [DavidS-Repo/chunker#3]
> pregen 10 1s world 1000b
[14:29:31 INFO]: Successfully loaded 3846 processed chunks for world
[14:29:31 INFO]: world radius reached. To process more chunks, please increase the radius.
> pregenoff
[14:29:38 INFO]: world pre-generator is already disabled.
> pregen 10 1s world 1000000c
[14:29:43 INFO]: world pre-generator is already enabled.

I cannot create new pre-generator before reload.

**Comment by DavidS-Repo** [comment]
I have patched the code to remove the task from the map upon completion or immediate termination. Also ensured `timerStart` is initialized before any possible termination to avoid miscalculated time outputs (e.g., “50+ years”).

```cmd
> pregen 16 1s world 1000b
[22:46:01 INFO]: Successfully loaded 176229 processed chunks for world
[22:46:01 INFO]: world radius reached. To process more chunks, please increase the radius.
[22:46:01 INFO]: [world]  Processed: 0     Chunks/s: 0     Completed: 176229         out of 3844           Chunks
[22:46:01 INFO]: Total time: 0 Seconds
> pregenoff
[22:46:12 INFO]: world pre-generator is already disabled.
> pregen 64 1s world 1000000c
[22:46:23 INFO]: Successfully loaded 176229 processed chunks for world
[22:46:23 INFO]: [world]  Processed: 50    Chunks/s: 50    Completed: 176285         out of 1000000000000  Chunks
[22:46:24 INFO]: [world]  Processed: 955   Chunks/s: 955   Completed: 177253         out of 1000000000000  Chunks
[22:46:25 INFO]: [world]  Processed: 765   Chunks/s: 765   Completed: 178021         out of 1000000000000  Chunks
[22:46:27 INFO]: [world]  Processed: 396   Chunks/s: 396   Completed: 178424         out of 1000000000000  Chunks
[22:46:28 INFO]: [world]  Processed: 250   Chunks/s: 250   Completed: 178680         out of 1000000000000  Chunks
```

This confirms the issue is now resolved. If you still encounter problems, please let me know.

The updated [Chunker-1.21.4.jar](https://github.com/DavidS-Repo/chunker/blob/main/Chunker-1.21.4.jar) version has the bug fix.

Thanks for reporting and contributing to improving the plugin!

**Comment by DerKloMensch** [comment]
Is it maybe possible for you to fix this also for 1.21.1 too? 👉👈
